tests/Avalonia.Controls.UnitTests/ComboBoxTests.cs

```diff
@@ -173,5 +176,67 @@ public void Detaching_Closed_ComboBox_Keeps_Current_Focus()
                 Assert.Equal(expectedSelectedIndex, target.SelectedIndex);
             }
         }
+        
+        [Fact]
+        public void SelectedItem_Validation()
+        {
+
+            using (UnitTestApplication.Start(TestServices.MockThreadingInterface))
+            {
+                var target = new ComboBox
+                {
+                    Template = GetTemplate()
+                };
+
+                target.ApplyTemplate();
+                target.Presenter.ApplyTemplate();
+                
+                var exception = new System.InvalidCastException("failed validation");
+                var textObservable = new BehaviorSubject<BindingNotification>(new BindingNotification(exception, BindingErrorType.DataValidationError));
+                target.Bind(ComboBox.SelectedItemProperty, textObservable);
+                
+                Dispatcher.UIThread.RunJobs();
+
+                Assert.True(DataValidationErrors.GetHasErrors(target));
+                Assert.True(DataValidationErrors.GetErrors(target).SequenceEqual(new[] { exception }));
+                
+            }
+            
+        }
+        private ComboBox CreateControl()
+        {
+            var control = new ComboBox()
+            {
+                Template = GetTemplate()
+            };
+
+            control.ApplyTemplate();
+            return control;
+        }
+        
+        private TextBox GetTextBox(ComboBox control)
+        {
+            return control.GetTemplateChildren()
+                // .OfType<ButtonSpinner>()
+                // .Select(b => b.Content)
+                .OfType<TextBox>()
+                .First();
+        }
+        // private IControlTemplate CreateTemplate()
```

Comment:
Is it WIP or can you remove this commented code?

Reply:
I will remove the code